src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/convert_matmul_to_fc.cpp

```diff
@@ -54,7 +54,7 @@ ov::intel_cpu::ConvertMatMulToFC::ConvertMatMulToFC() {
 
         // Transformation to FC is not supported for 1D inputs
         if (rank_a == 1 || rank_b == 1 ||
-            rank_a > 3 || rank_b > 3) {
+            rank_a > 4 || rank_b > 4) {
```

Comment:
It seems like we can totally remove upper limit for `rank_b`.
Further in this file is checked that weights path has no more than 2 dimensions not equal to 1. So the number of dimensions equal to one can be arbitrary

Reply:
Removed upper limit for rank_b